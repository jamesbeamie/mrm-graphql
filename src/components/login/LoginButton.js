import React from 'react';
import { loginButtonIcon } from '../../utils/images/images';

/**
 * Login Button component
 *
 * @returns {JSX}
 */
const LoginButton = () => {
  // get urls from the .env file
  const { ANDELA_LOGIN_URL, MRM_URL } = process.env || {};

  return (
    <a href={`${ANDELA_LOGIN_URL}=${MRM_URL}`}>
      <input
        type="image"
        src={loginButtonIcon}
        alt="Login With Your Email"
        className="btn-signin"
      />
    </a>
  );
};

export default LoginButton;
