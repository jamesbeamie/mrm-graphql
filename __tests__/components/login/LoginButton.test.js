import React from 'react';
import { shallow } from 'enzyme';
import LoginButton from '../../../src/components/login/LoginButton';

describe('LoginButton should', () => {
  it('should render properly', () => {
    global.process.env = {
      ANDELA_LOGIN_URL: 'mockLoginUrl',
      MRM_URL: 'mockMrmUrl',
    };
    expect(shallow(<LoginButton />)).toMatchSnapshot();
  });

  it('should render undefined when no environment variables are found', () => {
    global.process.env = null;
    expect(shallow(<LoginButton />)).toMatchSnapshot();
  });
});
