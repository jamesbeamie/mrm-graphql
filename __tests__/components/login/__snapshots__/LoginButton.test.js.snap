// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should should render properly 1`] = `
<a
  href="mockLoginUrl=mockMrmUrl"
>
  <input
    alt="Login With Your Email"
    className="btn-signin"
    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FButton.png?alt=media&token=fb7692b6-1465-47c5-94f6-c94aff31c275"
    type="image"
  />
</a>
`;

exports[`LoginButton should should render undefined when no environment variables are found 1`] = `
<a
  href="undefined=undefined"
>
  <input
    alt="Login With Your Email"
    className="btn-signin"
    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2FButton.png?alt=media&token=fb7692b6-1465-47c5-94f6-c94aff31c275"
    type="image"
  />
</a>
`;
